File: src/vehicle.c

#include <stdio.h>
#include <string.h>
#include "vehicle.h"

// Yeni araç oluşturur ve geri döndürür
Vehicle createVehicle(char licensePlate[], char ownerName[], char vehicleType[], int entryTime) {
    Vehicle v;
    strcpy(v.licensePlate, licensePlate);
    strcpy(v.ownerName, ownerName);
    strcpy(v.vehicleType, vehicleType);
    v.entryTime = entryTime;
    return v;
}

// Araç bilgilerini ekrana yazdırır
void printVehicle(Vehicle v) {
    printf("Plaka     : %s\n", v.licensePlate);
    printf("Sahip     : %s\n", v.ownerName);
    printf("Araç Tipi : %s\n", v.vehicleType);
    printf("Giriş     : %d\n", v.entryTime);
    printf("-------------------\n");
}

// İki aracın giriş saatini karşılaştırır
// v1 önce girdiyse -1, v2 önce girdiyse 1, aynıysa 0 döner
int compareEntryTime(Vehicle v1, Vehicle v2) {
    if (v1.entryTime < v2.entryTime) return -1;
    if (v1.entryTime > v2.entryTime) return 1;
    return 0;
}

File: src/vehicle.py
# Vehicle sınıfı - Araç bilgilerini saklar
class Vehicle:
    def __init__(self, license_plate, owner_name, vehicle_type, entry_time):
        self.license_plate = license_plate  # Plaka numarası
        self.owner_name = owner_name        # Araç sahibinin adı
        self.vehicle_type = vehicle_type    # Araç tipi
        self.entry_time = entry_time        # Giriş saati 

    def print_vehicle(self):
        print(f"Plaka     : {self.license_plate}")
        print(f"Sahip     : {self.owner_name}")
        print(f"Araç Tipi : {self.vehicle_type}")
        print(f"Giriş     : {self.entry_time}")
        print("-------------------")

    def compare_entry_time(self, other):
        if self.entry_time < other.entry_time:
            return -1
        elif self.entry_time > other.entry_time:
            return 1
        return 0